onnxruntime/core/providers/cpu/object_detection/roialign.cc

```diff
@@ -170,7 +170,8 @@ void RoiAlignForward(const TensorShape& output_shape, const T* bottom_data, floa
       int64_t index_n = n * channels * pooled_width * pooled_height;
 
       const T* offset_bottom_rois = bottom_rois + n * num_roi_cols;
-      const auto roi_batch_ind = batch_indices_ptr[n];
+      auto roi_batch_ind = batch_indices_ptr[n];
+      roi_batch_ind = (static_cast<uint64_t>(roi_batch_ind) < static_cast<uint64_t>(batch_size)) ? roi_batch_ind : 0;
```

Comment:
If batch index is out of bounds, shouldn't it just call it out as an error ?

Reply:
Hard error is certainly a valid choice, and one I also considered. I'm happy to change it, but I just wasn't sure what the impact might be to existing potentially invalid models (if they might start failing whereas before they may have silently succeeded and read some memory slightly beyond the end). Actually, let me split this into a separate PR from the warning, so it's much easier to decide in triage.

Reply:
Pulling out into separate PR.